### ISSUE TITLE
[4.x]: Super slow transaction query

### ISSUE THREAD

**Issue by samhibberd** [craftcms/commerce#3464]
### Description
Running a bunch of slow query testing on our site and found a really slow transaction query.

### Steps to reproduce

Slow query logging uncovered this query, run it with a valid hash:

```sql
SELECT `amount`, `code`, `currency`, `dateCreated`, `dateUpdated`, `gatewayId`, `hash`, `id`, `message`, `note`, `orderId`, `parentId`, `paymentAmount`, `paymentCurrency`, `paymentRate`, `reference`, `response`, `status`, `type`, `userId`
FROM `commerce_transactions`
WHERE `hash`='testestestestetstestestestestest'
ORDER BY `id`
LIMIT 1;
```

### Possible fix
Adding a `hash` index to the `commerce_transactions` table took it from 7s to under 10ms for us!

### Craft CMS version

4.8.9

### Craft Commerce version

4.5.3

### Installed plugins and versions

-

**Comment by samhibberd** [comment]
Awesome thanks 🎉 @nfourtythree

**Comment by nfourtythree** [comment]
Hi @samhibberd 

As it seems you have spotted this fix made it into version `4.6.0` of Commerce.

Thanks for raising the issue!
